python/pyarrow/memory.pxi

```diff
@@ -76,6 +76,12 @@ cdef class MemoryPool(_Weakrefable):
         """
         return frombytes(self.pool.backend_name())
 
+    def __repr__(self):
+        name = f"{self.__class__.__module__}.{self.__class__.__name__}"
+        return (f"{name}("
+                f"backend_name={self.backend_name}, "
+                f"bytes_allocated={self.bytes_allocated()}, "
+                f"max_memory={self.max_memory()})")
```

Comment:
Can we test this repr somewhere? (either a doctest or a pytest unit test)

Reply:
https://github.com/apache/arrow/pull/13921/commits/ae55f755a259b9079cb306fe224c98bcd93829b4